packages/token-bridge-sdk/src/hooks/useArbTokenBridge.ts

```diff
@@ -583,7 +588,7 @@ export const useArbTokenBridge = (
     const tx = await erc20Bridger.withdraw({
       l2Signer,
       erc20l1Address: erc20L1Address,
-      destinationAddress: walletAddress,
+      destinationAddress: destinationAddress ?? walletAddress,
```

Comment:
Not sure if necessary, but we could put an additional safety check here to guard against future footguns, i.e., check if L2 signer is a SC, if so, require destinationAddress

Reply:
it's technically not necessary because we are checking for it in the parent method calling our local sdk but I will add an additional safety check here as well

Reply:
actually I don't think we need to add this check here because we are doing all that in the bridge UI transfer methods and our sdk is just making the call. We don't have access to whether wallet is smart contract on the sdk level

Reply:
Actually, do you mean something like this?

```
    const provider = l2Signer.provider
    const isSmartContractAddress = provider && (await provider.getCode(String(erc20L1Address))).length < 2
    if (isSmartContractAddress && !destinationAddress) {
      throw new Error(`Missing destination address`)
    }
```

Reply:
^ yep, exactly. Even tho this is checked whenever the method is called in the codebase currently, the idea is to guard / insure against misuses in the future